homeassistant/components/shelly/sensor.py

```diff
@@ -104,6 +128,24 @@ def __init__(
         """Initialize select."""
         super().__init__(coordinator, key, attribute, description)
 
+        if hasattr(self, "_attr_name") and description.role != ROLE_GENERIC:
+            delattr(self, "_attr_name")
+
+        if not description.role:
+            translation_placeholders, translation_key = (
+                get_entity_translation_attributes(
+                    get_rpc_channel_name(coordinator.device, key),
+                    description.translation_key,
+                    description.device_class,
+                    self._default_to_device_class_name(),
+                )
+            )
+
+            if translation_placeholders:
+                self._attr_translation_placeholders = translation_placeholders
+                if translation_key:
+                    self._attr_translation_key = translation_key
```

Comment:
I think we don't need the if as those are anyway defined as `None` in the parent class isn't it ?